Amplify/Categories/DataStore/DataStoreCategoryBehavior.swift

```diff
@@ -15,51 +15,81 @@ public protocol DataStoreBaseBehavior {
     func save<M: Model>(_ model: M,
                         where condition: QueryPredicate?,
                         completion: @escaping DataStoreCallback<M>)
+    func save<M: Model>(_ model: M,
+                        where condition: QueryPredicate?) async -> DataStoreResult<M>
 
     @available(*, deprecated, message: "Use query(:byIdentifier:completion)")
     func query<M: Model>(_ modelType: M.Type,
                          byId id: String,
                          completion: DataStoreCallback<M?>)
+    @available(*, deprecated, message: "Use query(:byIdentifier:completion)")
```

Comment:
this should be 
```suggestion
    @available(*, deprecated, message: "Use query(:byIdentifier)")
```
the existing API is deprecated, and this PR adds a new async API of the deprecated one. removing the two deprecated APIs will be a separate effort